keras/applications/regnet.py

```diff
@@ -0,0 +1,1576 @@
+
+  The naming of models is as follows: `RegNet{block_type}{flops}` where 
+  `block_type` is one of `(Y, Z)` and `flops` signifies hundred million 
+  floating point operations. For example RegNetY64 corresponds to RegNet with 
+  Y block and 6.4 giga flops (64 hundred million flops). 
+
+  Args:
+    include_top: Whether to include the fully-connected
+        layer at the top of the network. Defaults to True.
+    weights: One of `None` (random initialization),
+          "imagenet" (pre-training on ImageNet),
+          or the path to the weights file to be loaded. Defaults to "imagenet".
+    input_tensor: Optional Keras tensor
+        (i.e. output of `layers.Input()`)
+        to use as image input for the model.
+    input_shape: Optional shape tuple, only to be specified
+        if `include_top` is False.
+        It should have exactly 3 inputs channels.
+    pooling: Optional pooling mode for feature extraction
+        when `include_top` is `False`. Defaults to None.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional layer.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional layer, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+    classes: Optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified. Defaults to 1000 (number of
+        ImageNet classes).
+    classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        Defaults to "softmax".
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+
+  Returns:
+    A `keras.Model` instance.
+"""
+
+
+def PreStem(name=None):
+  """
+  Rescales and normalizes inputs to [0,1] and ImageNet mean and std.
+  
+  Args:
+    name: name prefix
+    
+  Returns:
+    Rescaled and normalized tensor
+  """
+  if name is None:
+    name = "prestem" + str(backend.get_uid("prestem"))
+
+  def apply(x):
+    x = layers.Rescaling(scale=1. / 255., name=name + "_prestem_rescaling")(x)
+
+    return x
+
+  return apply
+
+
+def Stem(name=None):
+  """
+  Implementation of RegNet stem. (Common to all model variants)
+  
+  Args:
+    name: name prefix   
+
+  Returns:
+    Output tensor of the Stem
+  """
+  if name is None:
+    name = "stem" + str(backend.get_uid("stem"))
+
+  def apply(x):
+    x = layers.Conv2D(32, (3, 3),
+                      strides=2,
+                      use_bias=False,
+                      padding="same",
+                      kernel_initializer=initializers.HeNormal(),
```

Comment:
This can be replaced by `kernel_initializer="he_normal"` which might be a bit more readable.

This is more of a personal preference, so feel free to ignore.

Reply:
Sure, I'll replace that.